#!/usr/bin/env python
PACKAGE = "uam_arm"

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

gen.add("DEVICE", str_t, 0, "No. of ttyUSB", "/dev/ttyUSB0");
gen.add("servo_1", double_t, 0, "servo_1 position",  0.00, -3.14, 3.14);
gen.add("servo_2", double_t, 0, "servo_2 position",  1.05, -1.57, 1.57);
gen.add("servo_3", double_t, 0, "servo_3 position", -2.40, -2.60, 2.60);
gen.add("servo_4", double_t, 0, "servo_4 position", -1.65, -3.14, 3.14);
gen.add("servo_5", double_t, 0, "servo_5 position",  0.00, -1.57, 1.57);
gen.add("servo_6", double_t, 0, "servo_6 position",  0.09, -3.14, 3.14);
gen.add("Key", int_t, 0,"end_effector", 2, 0, 3);


gen.add("trim_1", double_t, 0, "servo_1 trim",  1.4444, -3.14, 3.14);
gen.add("trim_2", double_t, 0, "servo_2 trim",  -0.3768, -3.14, 3.14);
gen.add("trim_3", double_t, 0, "servo_3 trim",   0.628, -3.14, 3.14);
gen.add("trim_4", double_t, 0, "servo_4 trim",  -0.1884, -3.14, 3.14);
gen.add("trim_5", double_t, 0, "servo_5 trim",  -0.4396, -3.14, 3.14);
gen.add("trim_6", double_t, 0, "servo_6 trim",  -0.0628, -3.14, 3.14)

exit(gen.generate(PACKAGE, "uam_arm", "uam_"))
